src/components/banner/banner.ts

```diff
@@ -0,0 +1,148 @@
+
+export interface IgcBannerComponentEventMap {
+  igcClosing: CustomEvent<void>;
+  igcClosed: CustomEvent<void>;
+}
+
+/**
+ * The `igc-banner` component displays important and concise message(s) for a user to address, that is specific to a page or feature.
+ *
+ * @element igc-banner
+ *
+ * @slot - Renders the text content of the banner message.
+ * @slot prefix - Renders additional content at the start of the message block.
+ * @slot actions - Renders any action elements.
+ *
+ * @fires igcClosing - Emitted before closing the banner - when a user interacts (click) with the default action of the banner.
+ * @fires igcClosed - Emitted after the banner is closed - when a user interacts (click) with the default action of the banner.
+ *
+ * @csspart base - The base wrapper of the banner component.
+ * @csspart spacer - The inner wrapper that sets the space around the banner.
+ * @csspart message - The part that holds the text and the illustration.
+ * @csspart illustration - The part that holds the banner icon/illustration.
+ * @csspart content - The part that holds the banner text content.
+ * @csspart actions - The part that holds the banner action buttons.
+ */
+
+@themes(all)
+export default class IgcBannerComponent extends EventEmitterMixin<
+  IgcBannerComponentEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-banner';
+  public static styles = [styles];
+
+  public static register() {
+    registerComponent(IgcBannerComponent, IgcButtonComponent);
+  }
+
+  private _internals: ElementInternals;
+  private _bannerRef: Ref<HTMLElement> = createRef();
+  private _animationPlayer = addAnimationController(this, this._bannerRef);
+
+  /**
+   * Determines whether the banner is being shown/hidden.
+   * @attr
+   */
+  @property({ type: Boolean, reflect: true })
+  public open = false;
+
+  constructor() {
+    super();
+    this._internals = this.attachInternals();
+
+    this._internals.role = 'status';
+    this._internals.ariaLive = 'polite';
+  }
+
+  /** Shows the banner if not already shown. */
+  public show(): void {
+    if (this.open) {
+      return;
+    }
+
+    this.open = true;
+    this.toggleAnimation('open');
+  }
+
+  /** Hides the banner if not already hidden. */
+  public hide(): void {
+    if (!this.open) {
+      return;
+    }
+
+    this.toggleAnimation('close');
+    this.open = false;
+  }
+
+  /** Toggles between shown/hidden state. */
+  public toggle(): void {
+    this.open ? this.hide() : this.show();
+  }
+
+  private async toggleAnimation(dir: 'open' | 'close') {
+    const animation = dir === 'open' ? growVerIn : growVerOut;
+
+    const [_, event] = await Promise.all([
+      this._animationPlayer.stopAll(),
+      this._animationPlayer.play(animation()),
+    ]);
+
+    return event.type === 'finish';
+  }
+
+  private handleClick() {
+    const allowed = this.emitEvent('igcClosing', { cancelable: true });
+
+    if (allowed) {
```

Comment:
```suggestion
if (this.emitEvent('igcClosing', { cancelable: true })) {
```
Nit: You can shortcut that without the extra variable.